### PR TITLE
added lua module example closes #issue#7

### PR DESCRIPTION
**Pull request by waximabbax**
WIP: added lua module example closes #issue#7

### CODE REVIEW

**Comment by waximabbax** [comment]
manual test

image

**Comment by waximabbax** [comment]
parser tests

image

File: src/SparqlLua.php [hunk 1]
```diff
@@ -19,7 +19,7 @@ public function register(): array {
 		);
 	}
 
-	private function runSparqlQuery( string $sparql ): string|null {
+	private function runSparqlQuery( string $sparql ): array|null {
```
Comment (waximabbax) [inline src/SparqlLua.php:22]: if sparql is empty or wrong it returns null. we have a test for that as well.

File: src/SparqlLuaQueryRunner.php [hunk 2]
```diff
@@ -14,15 +14,16 @@ public function __construct(
...
 		$options = [
 			'postData' => [
 				'query' => $sparql,
 				'format' => 'json'
 			],
 		];
+		$response = $this->requestFactory->request( 'POST', $this->sparqlEndpoint, $options );
 
-		return $this->requestFactory->request( 'POST', $this->sparqlEndpoint, $options );
+		return $response !== null ? json_decode( $response, true ) : null;
```
Comment (waximabbax) [inline src/SparqlLuaQueryRunner.php:26]: if sparql is empty or wrong it $response is null. we have a test for that as well.

**Comment by JeroenDeDauw** [comment]
Is this ready for review?

**Comment by waximabbax** [comment]
@JeroenDeDauw  Its ready for the review now.

File: README.md [hunk 3]
```diff
@@ -30,12 +31,82 @@ Define a lua module that requires the `SPARQL` binding and uses its runQuery met
...
+local sparql = require('SPARQL')
+-- Import necessary modules
+local mwHtml = require('mw.html')
 
+-- Module definition
 local p = {}
 
-function p.runQuery(frame)
-	return sparql.runQuery(frame.args[1])[0] -- Use the runQuery method
+function reverseTable(t)
```
Comment (JeroenDeDauw) [inline README.md:41]: This can be local

File: tests/parser/lua.txt [hunk 4]
```diff
@@ -5,24 +5,101 @@ version=2
...
 
 !! test
 getLabel returns empty on subject not found
 !! config
 !! wikitext
-{{#invoke:Sparql|runQuery|sparql query be here}}
-!! html
+{{#invoke:MySPARQL|buildTableFromSparql|
+
+SELECT ?subject_col1 ?predicate ?object WHERE { ?subject_col1 ?predicate ?object } LIMIT 3
```
Comment (JeroenDeDauw) [inline tests/parser/lua.txt:99]: Ideally we have an easier-to-understand example. So with more meaningful value selection. This is tracked by https://github.com/ProfessionalWiki/MaRDI-Tasks/issues/5

**Pull request merged**